### ISSUE TITLE
TypeScript's "export import" is treated as import

### ISSUE THREAD

**Issue by be5invis** [benmosher/eslint-plugin-import#1829]
TypeScript has a special syntax to re-export an namespace looks like `export import <name> = <binding>`. Currently eslint-plugin-import is treating them as imports and applies incorrect rules.

```ts
// with import/order, the "export import" line will cause an error
import { Point } from "./point";
export namespace SomeNamespace {
    export import Pt = Point;
}
```

**Comment by ljharb** [comment]
If you could provide a PR with a failing test case, that'd be most helpful.

**Comment by ljharb** [comment]
Although, this might have been fixed in #1823 - can you check with latest master?

**Comment by be5invis** [comment]
@ljharb It looks like that #1823 fixed this issue. I created a PR to add a test about this.

**Comment by be5invis** [comment]
@ljharb Tests in #1823 shows `newline-after-import` doesn't recognize TypeScript's `export import` and treated it as an import instead of an export.

**Comment by ljharb** [comment]
Fixed in #1830.
